src/Docker.DotNet/Endpoints/SwarmOperations.cs

```diff
@@ -15,7 +15,8 @@ internal class SwarmOperations : ISwarmOperations
         {
             if (statusCode < HttpStatusCode.OK || statusCode >= HttpStatusCode.BadRequest)
             {
-                throw new DockerApiException(statusCode, responseBody);
+                var deserializedBody = new JsonSerializer().DeserializeObject<JSONError>(responseBody);

```

Comment:
Use `this._client.JsonSerializer` instead of `new JsonSerializer()` so that the user settings apply to this as well.

Reply:
It's not accessible from the delegate, unless we change the ApiResponseErrorHandlingDelegate to take jsonserializer as a parameter, which will affect other handlers. It would make sense since all exceptions from docker api have json in the response body. Maybe it's something to consider in future improvments?

Reply:
It could be made non-static, I think...

Reply:
As far I know it cannot be made non-static, but I might be wrong?

Reply:
What is wrong with this?

![image](https://user-images.githubusercontent.com/4714040/48939690-30a3a380-eefc-11e8-90c7-f6eef6213cc0.png)

Reply:
The problem is that you can't access "this", because it's an anonymous method. Try initialize swarmresponsehandler with a non-static method in Test, and you will get compilation error.

Reply:
My mistake, it's working now. Actually I got the "A field initializer cannot reference the nonstatic field, method, or property" and thought it had to do with the non-static method, and it sort of had, but the problem was that the delegate had to be instanciated in the constructor.

Thanks for your patience.